src/shared/mount-util.c

```diff
@@ -77,13 +77,25 @@ int umount_recursive(const char *prefix, int flags) {
         return n;
 }
 
-static int get_mount_flags(const char *path, unsigned long *flags) {
-        struct statvfs buf;
+/* Get the mount flags for the mountpoint at "path" from "table" */
+static int get_mount_flags(const char *path, unsigned long *flags, struct libmnt_table *table) {
+        struct libmnt_fs *fs = NULL;
+        const char *opts = NULL;
+        int r = 0;
+
+        fs = mnt_table_find_target(table, path, MNT_ITER_FORWARD);
+        if (!fs) {
+            r = -ENOENT;
+        } else {
+            opts = mnt_fs_get_vfs_options(fs);
+            r = mnt_optstr_get_flags(opts, flags, mnt_get_builtin_optmap(MNT_LINUX_MAP));
+            /* relatime is default and trying to set it in an unprivileged container causes EPERM */
+            *flags &= ~MS_RELATIME;
+        }
+        if (r)
```

Comment:
we only use c's downgrade-to-bool feature for stuff that is either actually a bool (or at least boolean-like) or a pointer. For values that are actually used with a larger range of numeric values (like this one), we avoid it, to keep this more readable. See: 

https://systemd.io/CODING_STYLE#using-c-constructs